File: Problemset/0405-Convert a Number to Hexadecimal/Bitwise Operator.py
class Solution:
    def toHex(self, num: int) -> str:
        hex_ch = "0123456789abcdef"
        hex_n = ""

        if num == 0:
            return "0"

        if num < 0:
            num &= 0x7fffffff
            hex_n += hex_ch[num & 0xf]
            num >>= 4
            num |= 0x08000000

        while num != 0:
            hex_n += hex_ch[num & 0xf]
            num >>= 4

        return hex_n[::-1]

File: Problemset/0476-Number Complement/Bitwise Operator.rs
impl Solution {
    pub fn find_complement(num: i32) -> i32 {
        let mut ret = 0;

        for i in 0..30 {
            if num >> i == 0 {
                break;
            }
            if num & (1 << i) == 0 {
                ret ^= 1 << i;
            }
        }

        ret
    }
}
